conans/client/rest/remote_credentials.py

```diff
@@ -4,16 +4,29 @@
 
 from jinja2 import Template
 
+from conan.internal.cache.home_paths import HomePaths
+
+from conans.client.loader import load_python_file
 from conan.api.output import ConanOutput
 from conans.client.userio import UserInput
-from conans.errors import ConanException
+from conans.errors import ConanException, scoped_traceback
 from conans.util.files import load
 
+_default_auth_plugin = """
+# This file was generated by Conan. Remove this comment if you edit this file or Conan
+# will destroy your changes.
+
+def auth_plugin(remote, user=None, password=None):
```

Comment:
That fact that this is only the `remote.name` might hamper users from coming up with useful solutions, and they have no (easy) way to instantiate the proper `conan_api` either to re-resolve it. Maybe we can check if passing the whole `Remote` object would be better?